urllib3/connection.py

```diff
@@ -162,6 +162,27 @@ def connect(self):
         conn = self._new_conn()
         self._prepare_conn(conn)
 
+    def request_chunked(self, method, url, body=None, headers={}):
+        """ 
+        Alternative to the common request method, which sends the 
+        body with chunked encoding and not as one block 
+        """
+        header_names = set(k.lower() for k in headers)
+        self.putrequest(method, url, skip_accept_encoding=True)
+        for header, value in headers.items():
+            self.putheader(header, value)
+        if 'transfer-encoding' not in headers:
+            self.putheader('Transfer-Encoding', 'chunked')
```

Comment:
Hmm. I'm a little nervous about this. There are a few other transfer-encodings defined for HTTP (compress, deflate, and gzip are all defined in RFC 7230. It's *possible*, albeit unlikely, that users may be attempting to use one of those transfer encodings *and* want us to chunk-encode it. If they did, we'd get this wrong.

I think the correct thing to look for is whether `chunked` is in any of the values of the `transfer-encoding` header, and if not to add it by appending to it. That's a little more complex and we may not need to do it if we're happy with the edge-case.